pageserver/src/tenant/metadata.rs

```diff
@@ -267,23 +269,32 @@ pub fn save_metadata(
     Ok(())
 }
 
+#[derive(Error, Debug)]
+pub enum LoadMetadataError {
+    #[error("Got io error while reading metadata at {path}: {source}")]
+    Read { source: io::Error, path: Box<str> },
```

Comment:
I don't think we really need the path at all here; I should had pushed back when they were earlier added to a lot of errors.

We don't we need it here: `PageserverConf::metadata_path()` is a pure function and I can deduce it fast enough (`$repo/tenants/{tenant_id}/{timeline_id}/metadata`), only config relative is the `$repo` path. Though, that's a lot of `PathBuf::join`.

Reply:
So the suggestion is to add `conf` reference to the error and plumb the path to be showed on demand in Display impl?

We need the path anyway in the load function, so dropping it there and recreating it at a later stage doesnt make a real difference IMO

Maybe the solution is to work with `LoadLocalTimelineError`, i e include timeline id there or something. So upper error reports timeline id and by timeline id you know where the metadata is supposed to be, but thats a larger refactoring

Reply:
(Already chatted about this but) my suggestion is to just drop the `path` props from this type. Let's add a `.with_context(format!("load metadata for {timeline_id}")` at the caller.